control-base/src/main/java/fi/nls/oskari/util/WFSGetLayerFields.java

```diff
@@ -0,0 +1,157 @@
+                        attributeValue instanceof Integer ||
+                        attributeValue instanceof Long ||
+                        attributeValue instanceof Float ||
+                        attributeValue instanceof Double
+                    ) {
+                        attributeType = NUMBER;
+                    }
+
+                    // update attribute type if its not set or UNKNOWN
+                    String currentAttributeType = attributes.optString(attributeName);
+                    if (currentAttributeType.isEmpty() || currentAttributeType.equals(UNKNOWN)) {
+                        attributes.put(attributeName, attributeType);
+                    }
+                }
+            }
+            final JSONObject result = new JSONObject();
+            result.put(ATTRIBUTES_KEY, attributes);
+            result.put(GEOMETRY_FIELD_KEY, "geometry");
+            return result;
+        } catch (IOException ex) {
+            throw new ServiceException("Couldn't reach url: " + url);
+        } catch (JSONException ex) {
+            throw new ServiceException("Couldn't parse the result", ex);
+        }
+    }
+
+    /**
+     * Return fields information for WFS 1.x/2.x layer
+     *
+     * The fields are collected by parsing the DescribeFeatureType
+     * response. The xml types are converted to STRING, NUMBER,
+     * BOOLEAN and UNKNOWN to conform to the same set of types
+     * as WFS 3.x
+     *
+     */
+    private static JSONObject getFeatureTypeFields(OskariLayer layer) throws ServiceException {
+        final JSONObject response = WFSDescribeFeatureHelper.getWFSFeaturePropertyTypes(layer, String.valueOf(layer.getId()));
+        final Set<String> geometryPropertyTypes = new HashSet<>(
+            Arrays.asList(
+                "GeometryPropertyType",
+                "PointPropertyType",
+                "LinePropertyType",
+                "PolygonPropertyType",
+                "MultiPointPropertyType",
+                "MultiLinePropertyType",
+                "MultiPolygonPropertyType"
+            )
+        );
+        final Set<String> stringTypes = new HashSet<>(Arrays.asList("string", "date", "time"));
+        final Set<String> numericTypes = new HashSet<>(Arrays.asList("decimal", "int", "integer", "float", "double"));
+        final Set<String> booleanTypes = new HashSet<>(Arrays.asList("boolean"));
+        try {
+            final JSONObject propertyTypes = response.getJSONObject("propertyTypes");
+            final Iterator<?> attributeNames = propertyTypes.keys();
+            final JSONObject attributes = new JSONObject();
+            final JSONObject result = new JSONObject();
+            while (attributeNames.hasNext()) {
+                String attributeName = (String) attributeNames.next();
+                String attributeType = propertyTypes.getString(attributeName);
+                attributeType = attributeType.split(":")[1];  // remove xml namespace prefix
```

Comment:
We should make sure there is a ':' character included and not assume there is. Most probably have it but there's always the chance that XML default namespace is set based on the featuretype.